lib/useOnyx.ts

```diff
@@ -0,0 +1,189 @@
+import {deepEqual} from 'fast-equals';
+import {useCallback, useEffect, useRef, useSyncExternalStore} from 'react';
+import Onyx from './Onyx';
+import type {CollectionKeyBase, OnyxCollection, OnyxKey, OnyxValue, Selector} from './types';
+import useLiveRef from './useLiveRef';
+import usePrevious from './usePrevious';
+
+type UseOnyxOptions<TKey extends OnyxKey, TReturnValue> = {
+    /**
+     * Determines if this key in this subscription is safe to be evicted.
+     */
+    canEvict?: boolean;
+
+    /**
+     * If set to false, then no data will be prefilled into the component.
+     */
+    initWithStoredValues?: boolean;
+
+    /**
+     * If set to true, data will be retrieved from cache during the first render even if there is a pending merge for the key.
+     */
+    allowStaleData?: boolean;
+
+    /**
+     * This value will be returned by the hook on the first render while the data is being read from Onyx.
+     */
+    initialValue?: OnyxValue<TKey>;
+
+    /**
+     * This will be used to subscribe to a subset of an Onyx key's data.
+     * Using this setting on `useOnyx` can have very positive performance benefits because the component will only re-render
+     * when the subset of data changes. Otherwise, any change of data on any property would normally
+     * cause the component to re-render (and that can be expensive from a performance standpoint).
+     */
+    selector?: Selector<TKey, unknown, TReturnValue>;
+};
+
+type FetchStatus = 'loading' | 'loaded';
```

Comment:
let's use an object / const here and export it so we can check loading status with a const in E/App?

Reply:
sounds good